querybook/webapp/ui/Comment/Comments.tsx

```diff
@@ -1,184 +1,216 @@
+        (parentCommentId: number, childCommentIds: number[] = []) => {
+            dispatch(
+                fetchChildCommentsByParentCommentIdIfNeeded(
+                    parentCommentId,
+                    childCommentIds
+                )
+            );
+            setOpenThreadIds((curr) => new Set([...curr, parentCommentId]));
+        },
+        [dispatch]
+    );
+
+    const loadingCommentDOM = React.useMemo(
+        () => (
+            <div className="Comment mv8">
+                <StyledText
+                    size="xsmall"
+                    color="lightest"
+                    cursor="default"
+                    isItalic
                 >
+                    Loading Comment
+                </StyledText>
+            </div>
+        ),
+        []
+    );
+
+    const renderThreadCommentDOM = (comment: IComment) => (
+        <>
+            {renderFlatCommentDOM(comment, false)}
+            <div
+                className="CommentThread mt16 mb12"
+                onClick={() =>
+                    handleOpenThread(comment.id, comment.child_comment_ids)
+                }
+            >
+                {openThreadIds.has(comment.id) ? (
+                    comment.child_comment_ids?.map((commentId) =>
+                        commentsById[commentId]
+                            ? renderFlatCommentDOM(
+                                  commentsById[commentId],
+                                  true
+                              )
+                            : loadingCommentDOM
+                    )
+                ) : (
                     <div className="ClosedCommentThread flex-row">
-                        <div className="flex-row mr8">
-                            {uids
-                                .slice(0, MAX_THREAD_AVATAR_COUNT)
-                                .map((uid) => (
-                                    <UserAvatar
-                                        key={`thread-avatar-${uid}-${comment.id}`}
-                                        uid={uid}
-                                        tiny
-                                    />
-                                ))}
-                            {uids.length - 1 >= MAX_THREAD_AVATAR_COUNT ? (
-                                <div className="NumberAvatar">
-                                    {uids.length - MAX_THREAD_AVATAR_COUNT}
-                                </div>
-                            ) : null}
-                        </div>
                         <StyledText
                             className="ThreadCount mr8"
                             size="xsmall"
                             color="accent"
                             cursor="default"
                         >
-                            {comments.length}{' '}
-                            {comments.length === 1 ? 'Reply' : 'Replies'}
+                            {comment.child_comment_ids.length}{' '}
```

Comment:
you checked `comment.child_comment_ids?.map` with `?` above, while not checking here. could you double confirm if it's really needed and make them consistent?